x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/allocation/decider/HasFrozenCacheAllocationDecider.java

```diff
@@ -83,7 +95,21 @@ private Decision canAllocateToNode(IndexMetadata indexMetadata, DiscoveryNode di
             case HAS_CACHE -> HAS_FROZEN_CACHE;
             case NO_CACHE -> NO_FROZEN_CACHE;
             case FAILED -> UNKNOWN_FROZEN_CACHE;
-            default -> STILL_FETCHING;
+            // THROTTLE is an odd choice here. In all other places, it means YES but not right now. But here it can eventually
+            // turn into a NO if the node responds that it has no frozen cache. Since BalancedShardAllocator effectively handles
+            // THROTTLE as a rejection, it is simpler that we just return a NO here for shard level decisions. For BWC, we keep
```

Comment:
If BalancedShardAllocator does treat THROTTLE as NO, then can't we replace THROTTLE in non-simulation?

I'm concerned the reasoning here is a bit hand-wavy. If we don't think THROTTLE makes sense here, we should remove it. Or explain why we can't.

Reply:
Err I should have flipped this PR to draft. These are kinda experimental code for investigating Nick's suggestion. Let me do that and I'll tag again when it is ready.